src/Microsoft.Extensions.ServiceDiscovery/Configuration/ConfigurationServiceEndpointProvider.cs

```diff
@@ -135,46 +127,60 @@ public ValueTask PopulateAsync(IServiceEndpointBuilder endpoints, CancellationTo
             }
         }
 
-        // Filter the resolved endpoints to only include those which match the specified scheme.
-        var minIndex = _schemes.Length;
-        foreach (var ep in resolved)
+        int resolvedEndpointCount;
+        if (_includeAllSchemes)
         {
-            if (ep.EndPoint is UriEndPoint uri && uri.Uri.Scheme is { } scheme)
+            // Include all endpoints.
+            foreach (var ep in resolved)
             {
-                var index = Array.IndexOf(_schemes, scheme);
-                if (index >= 0 && index < minIndex)
-                {
-                    minIndex = index;
-                }
+                endpoints.Endpoints.Add(ep);
             }
+            
+            resolvedEndpointCount = resolved.Count;
         }
-
-        var added = 0;
-        foreach (var ep in resolved)
+        else
         {
-            if (ep.EndPoint is UriEndPoint uri && uri.Uri.Scheme is { } scheme)
+            // Filter the resolved endpoints to only include those which match the specified, allowed schemes.
+            resolvedEndpointCount = 0;
+            var minIndex = _schemes.Length;
+            foreach (var ep in resolved)
             {
-                var index = Array.IndexOf(_schemes, scheme);
-                if (index >= 0 && index <= minIndex)
+                if (ep.EndPoint is UriEndPoint uri && uri.Uri.Scheme is { } scheme)
                 {
-                    ++added;
-                    endpoints.Endpoints.Add(ep);
+                    var index = Array.IndexOf(_schemes, scheme);
+                    if (index >= 0 && index < minIndex)
+                    {
+                        minIndex = index;
```

Comment:
Re-reading this logic, not sure I understand it.
e.g.
`_schemes = [ "https", "http" ]`
`resolved = [ "http://basket", "https://orders" ]`
`minIndex` will be 0 so we'll only add `"https://orders"` to `endpoints` below?

Reply:
Yes, that's correct. `_schemes` is in order of preference, lower index = higher preference. We want to pick preferred results only: if an `https` result is found, exclude `http` results, since `https` appears before `http` in `_schemes`.

Reply:
The idea is to allow `https+http://foo` so that we can use `http` at dev time and restrict ourselves to `https` in prod using config without changing the code. We changed our policy and opted to use HTTPS everywhere, when possible, but it's still not straightforward/automated on some Linux distros, so there is still some value to this otherwise whacky feature.

Reply:
> But it's applying to different endpoints? 

I don't quite understand this, what does it mean?

Reply:
Offline discussion: This isn't a mix of endpoints, so my question doesn't apply.